### PR TITLE
[DCWL-2275] Detach customs-api-common & Play 3.0 Upgrade

### PR DESCRIPTION
**Pull request by JakeReid2020**
Please ensure we are following our Ways of Working for Pull Requests: https://confluence.tools.tax.service.gov.uk/display/BTL/Ways+of+Working+for+Pull+Requests

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -2,21 +2,23 @@ import sbt._
 
 object AppDependencies {
```
Comment (IanWood-HMRC) [inline project/AppDependencies.scala:3]:
```suggestion
object AppDependencies {

  val playSuffix       = "-play-30"
  val bootstrapVersion = "8.0.0"
  val hmrcMongoVersion = "1.8.0"
  val scalamockVersion = "5.2.0"

  val compile = Seq(
    "uk.gov.hmrc"        %% s"bootstrap-backend$playSuffix" % bootstrapVersion,
    "org.typelevel"      %% "cats-core"                     % "2.10.0",
    "uk.gov.hmrc.mongo"  %% s"hmrc-mongo$playSuffix"        % hmrcMongoVersion,
    "com.typesafe.play"  %% "play-json-joda"                % "2.8.1"
  )

  val test = Seq(
    "org.scalatestplus"  %% "scalatestplus-mockito"        % "1.0.0-M2" ,
    "uk.gov.hmrc"        %% s"bootstrap-test$playSuffix"   % bootstrapVersion,
    "uk.gov.hmrc.mongo"  %% s"hmrc-mongo-test$playSuffix"  % hmrcMongoVersion,
    "org.scalamock"      %% "scalamock"                    % scalamockVersion
  ).map(_ % Test)
}

```

You could removed some spaces and map the Test bit.

**Comment by IanWood-HMRC** [comment]
You may be able to change some versions

[info] Found 7 dependency updates for customs-declarations-metrics
[info]   com.typesafe.play:play-json-joda               : 2.8.1   -> 2.8.2   -> 2.10.4         
[info]   org.scala-lang:scala-library                   : 2.13.12 -> 2.13.13                   
[info]   org.scalamock:scalamock:test                   : 5.2.0                        -> 6.0.0
[info]   uk.gov.hmrc.mongo:hmrc-mongo-play-30           : 1.8.0              -> 1.9.0          
[info]   uk.gov.hmrc.mongo:hmrc-mongo-test-play-30:test : 1.8.0              -> 1.9.0          
[info]   uk.gov.hmrc:bootstrap-backend-play-30          : 8.0.0              -> 8.5.0          
[info]   uk.gov.hmrc:bootstrap-test-play-30:test        : 8.0.0              -> 8.5.0

**Comment by IanWood-HMRC** [comment]
You could also run Optimize Imports on **app** and **test**.

File: app/uk/gov/hmrc/customs/declarations/metrics/common/config/ConfigValidatedNelAdaptor.scala [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    def string(key: String): CustomsValidatedNel[String]
+    def int(key: String): CustomsValidatedNel[Int]
+    def boolean(key: String): CustomsValidatedNel[Boolean]
+    def duration(key: String): CustomsValidatedNel[Duration]
+  }
+
+  trait RootValidatedNelAdaptor extends ValidatedNelAdaptor {
+    def maybeString(key: String): CustomsValidatedNel[Option[String]]
+    def maybeBoolean(key: String): CustomsValidatedNel[Option[Boolean]]
+    def stringSeq(key: String): CustomsValidatedNel[Seq[String]]
```
Comment (gavinharris43) [inline app/uk/gov/hmrc/customs/declarations/metrics/common/config/ConfigValidatedNelAdaptor.scala:72]:
```suggestion
    def string(key: String): CustomsValidatedNel[String]
    def int.     (key: String): CustomsValidatedNel[Int]
    def boolean(key: String): CustomsValidatedNel[Boolean]
    def duration(key: String): CustomsValidatedNel[Duration]
  }

  trait RootValidatedNelAdaptor extends ValidatedNelAdaptor {
    def maybeString(key: String): CustomsValidatedNel[Option[String]]
    def maybeBoolean(key: String): CustomsValidatedNel[Option[Boolean]]
    def stringSeq(key: String): CustomsValidatedNel[Seq[String]]
```

**Review by gavinharris43 — DISMISSED** [review]

**Review by IanWood-HMRC — APPROVED** [review]

**Pull request merged**